packages/babel-core/src/config/files/configuration.js

```diff
@@ -24,6 +24,30 @@ const BABELRC_FILENAME = ".babelrc";
 const BABELRC_JS_FILENAME = ".babelrc.js";
 const BABELIGNORE_FILENAME = ".babelignore";
 
+export function findClosestConfig(rootDir: string): string {
+  let dirname = rootDir;
+  while (true) {
+    if (fs.existsSync(path.join(dirname, BABEL_CONFIG_JS_FILENAME))) {
+      return dirname;
+    }
+
+    const nextDir = path.dirname(dirname);
+    if (dirname === nextDir) break;
+    dirname = nextDir;
+  }
+
+  throw Object.assign(
+    (new Error(
+      `Babel was run with autoRoot:true but a root babel.config.js could not ` +
+        `be found when searching from "${rootDir}"`,
```

Comment:
To help the user troubleshoot its setup, we could log every dir?
> [...] when searching in `/babel/a`, `/babel`, `/`, ...

Reply:
I tweaked the text to mention the upward search, do you think that's enough? I worry that including all the directories would make the error harder to read.